components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -41,6 +50,22 @@ class DownloadsFeature(
      */
     fun start() {
         super.observeSelected()
+        fun reAttachOnStartDownloadListener(previousDialog: DownloadDialogFragment?) {
+            previousDialog?.apply {
+                this@DownloadsFeature.dialog = this
+                val selectedSession = sessionManager.selectedSession
+                val download = selectedSession?.download
+                download?.consume {
+                    onDownload(selectedSession, it)
+                    false
+                }
+            }
+        }
+
+        if (isAlreadyADialogCreated()) {
+            val previousDialog = getPreviousDialog()
+            reAttachOnStartDownloadListener(previousDialog)
```

Comment:
`isAlreadyADialogCreated()` seems unnecessary here as it will also call `getPreviousDialog()`. In Kotlin with nullability I'd expect something more like:

```Kotlin
findPreviousDialogFragment()?.let { attachToDialogFragment(it) }
```

Reply:
👍

Reply:
Fixed!